### PR TITLE
Tqct 558 public api docs for new regex check 2

### PR DESCRIPTION
**Pull request by alubenskyi**

### CODE REVIEW

File: spec/openapi.yaml [hunk 1]
```diff
@@ -10663,6 +10663,12 @@ paths:
   /translation-quality-api/v2/dictionary/check-types/TRADEMARK_CONSISTENCY/trademark-types:
     $ref: './spec/translation_quality/dictionaries_result.yaml#/x-paths/dictionary_checkType_TRADEMARK_CONSISTENCY_trademark_types'
 
+  /translation-quality-api/v2/dictionary/check-types/REGEX_CHECK/rule-types:
+    $ref: './spec/translation_quality/dictionaries.yaml#/x-paths/dictionary_regex_check_rule_types'
```
Comment (achebanets) [inline spec/openapi.yaml:10667]: dictionaries_result.yaml

File: spec/openapi.yaml [hunk 2]
```diff
@@ -10663,6 +10663,12 @@ paths:
...
+
+  /translation-quality-api/v2/dictionary/check-types/REGEX_CHECK/flavors:
+    $ref: './spec/translation_quality/dictionaries.yaml#/x-paths/dictionary_regex_check_flavors'
+
```
Comment (achebanets) [inline spec/openapi.yaml:10671]: dictionaries_result.yaml

File: spec/openapi.yaml [hunk 3]
```diff
@@ -10720,6 +10726,30 @@ paths:
...
     $ref: './spec/translation_quality/check_type_severity_level_result.yaml#/x-paths/profiles_check_type_group_severity_level'
 
+  /translation-quality-api/v2/accounts/{accountUid}/profiles/{profileUid}/check-types/REGEX_CHECK/settings/rules:
+    $ref: './spec/translation_quality/regex_check_settings.yaml#/x-paths/profiles_regex_check_type_rules'
+
+  /translation-quality-api/v2/accounts/{accountUid}/profiles/{profileUid}/check-types/REGEX_CHECK/settings/rules/{ruleUid}:
+    $ref: './spec/translation_quality/regex_check_settings.yaml#/x-paths/profiles_regex_check_type_rule'
+
+  /translation-quality-api/v2/accounts/{accountUid}/profiles/{profileUid}/check-types/REGEX_CHECK/settings/rules/{ruleUid}/severity-level:
+    $ref: './spec/translation_quality/regex_check_settings.yaml#/x-paths/profiles_regex_check_type_rule_severity_level'
```
Comment (achebanets) [inline spec/openapi.yaml]: If you add "makeResultForCertainFileFunction" for the file  it needs to change the file name to the regex_check_settings_result.yaml

File: spec/translation_quality/tqc_common.yaml [hunk 4]
```diff
@@ -102,6 +102,13 @@ components:
       description: offset number.
       example: 1
 
+    regexRuleUidPathParameter:
+      name: ruleUid
+      in: path
+      required: true
+      type: string
+      description: Unique identifier of corresponding check rule.
+
```
Comment (achebanets) [inline spec/translation_quality/tqc_common.yaml:111]: "A unique check rule identifier"?

File: spec/translation_quality/tqc_common.yaml [hunk 5]
```diff
@@ -164,6 +171,116 @@ components:
           type: string
           description: Unique identifier of corresponding group.
```
Comment (achebanets) [inline spec/translation_quality/tqc_common.yaml:172]: The same as above?

File: spec/translation_quality/tqc_common.yaml [hunk 6]
```diff
@@ -164,6 +171,116 @@ components:
...
+        - originalPattern
+        - translationPattern
+        - caseInsensitive
+
+    regexRuleWithRuleUid:
+      allOf:
+        - type: object
+          properties:
+            ruleUid:
+              type: string
```
Comment (achebanets) [inline spec/translation_quality/tqc_common.yaml:254]: inject from common file

Comment (alubenskyi) [reply spec/translation_quality/tqc_common.yaml:254]: it's already in tqc common file I don't prefer to move it into global common file

**Review by achebanets — APPROVED** [review]

File: spec/translation_quality/dictionaries.yaml [hunk 7]
```diff
@@ -232,6 +232,59 @@ x-paths:
...
+      produces:
+        - application/json
+
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DictionaryRegexCheckRuleTypesResponse'
```
Comment (rischenko) [inline spec/translation_quality/dictionaries.yaml:253]: why do we use both camel case and snake case?

Comment (rischenko) [reply spec/translation_quality/dictionaries.yaml:253]: $ref: '#/components/schemas/DictionaryRegexCheckRuleTypesResponse'
$ref: '#/components/examples/regex_check_rule_types_example'

File: spec/translation_quality/dictionaries_result.yaml [hunk 8]
```diff
@@ -284,6 +285,13 @@ x-paths:
                           apply to languages that only require one plural form
                           (Chinese, Japanese, etc.).
                         checkSubTypes: []
+                      - checkTypeCode: REGEX_CHECK
+                        name: Custom Quality Checks
+                        description: >-
+                          Custom quality checks that have been created by users.
+                          Click "new" button to add new ones or "edit" to modify
```
Comment (rischenko) [inline spec/translation_quality/dictionaries_result.yaml:292]: Click "new" button - looks strange for api documentation

**Review by rischenko — APPROVED** [review]

**Pull request merged**